state-chain/pallets/cf-pools/src/lib.rs

```diff
@@ -971,10 +973,14 @@ pub mod pallet {
 		#[pallet::weight(T::WeightInfo::set_maximum_price_impact())]
 		pub fn set_maximum_price_impact(
 			origin: OriginFor<T>,
+			base_asset: Asset,
+			quote_asset: Asset,
 			ticks: Option<u32>,
 		) -> DispatchResult {
 			T::EnsureGovernance::ensure_origin(origin)?;
-			MaximumPriceImpact::<T>::set(ticks);
+
+			let asset_pair = AssetPair::try_new::<T>(base_asset, quote_asset)?;
+			MaximumPriceImpact::<T>::mutate(asset_pair, |limit| *limit = ticks);
```

Comment:
Let's also add an event `PriceImpactLimitSet { ticks }`.

(Also we can just set the value, we don't need to mutate).

Reply:
Since the input ticks is Option<u32>, (support the case where we want to remove the limit), with set(), we have to consider the case where the input is None.
```
match ticks {
	Some(tick) => MaximumPriceImpact::<T>::insert(asset_pair, tick),
	None => MaximumPriceImpact::<T>::remove(asset_pair),
}
```

I just thought that using 1 line of mutate() looks better

Reply:
There is also a method `set`: `MaximumPriceImpact::<T>::set(asset_pair, ticks)` which does exactly this :)